.github/workflows/ci.yml

```diff
@@ -0,0 +1,21 @@
+name: CI
+on:
+  - push
+  - pull_request
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+      - uses: actions/setup-java@cd89f46ac9d01407894225f350157564c9c7cee2 # v3.12.0
```

Comment:
Just out of curiosity, why pinning the versions down to the commit hash? Security?

Reply:
@marchof yes - easy to tamper/overwrite existing tag, but much harder to tamper commit hash.